search-service/src/test/kotlin/com/egm/stellio/search/service/EntityPayloadServiceTests.kt

```diff
@@ -188,6 +191,67 @@ class EntityPayloadServiceTests : WithTimescaleContainer, WithKafkaContainer {
         }
     }
 
+    @Test
+    fun `it should replace an entity payload from an NGSI-LD Entity if existed yet`() = runTest {
+        val (jsonLdEntity, ngsiLdEntity) = loadSampleData().sampleDataToNgsiLdEntity().shouldSucceedAndResult()
+        entityPayloadService.createEntityPayload(ngsiLdEntity, now, jsonLdEntity)
+            .shouldSucceed()
+
+        entityPayloadService.replaceEntityPayload(ngsiLdEntity, now, jsonLdEntity)
+            .shouldSucceed()
+    }
+
+    @Test
+    fun `it should replace entity`() = runTest {
+        val beehiveURI = "urn:ngsi-ld:BeeHive:TESTC".toUri()
+        coEvery {
+            temporalEntityAttributeService.createEntityTemporalReferences(any(), any(), any(), any(), any())
+        } returns Unit.right()
+
+        val createEntityPayload = loadSampleData("beehive_minimal.jsonld")
+
+        entityPayloadService.createEntity(
+            createEntityPayload,
+            listOf(APIC_COMPOUND_CONTEXT),
+            "0123456789-1234-5678-987654321"
+        ).shouldSucceed()
+
+        entityPayloadService.retrieve(beehiveTestCId)
+            .shouldSucceedWith {
+                assertEquals(listOf(APIC_COMPOUND_CONTEXT), it.contexts)
+            }
```

Comment:
is it useful for the current test?

Reply:
but it can be useful to check the entity before the replace

Reply:
I'm sure it is already checked in another test